src/de.rs

```diff
@@ -140,6 +155,54 @@ impl<'de, R: Read<'de>> Deserializer<R> {
         }
     }
 
+    /// Parse arbitrarily deep JSON structures without any consideration for
+    /// overflowing the stack.
+    ///
+    /// You will want to provide some other way to protect against stack
+    /// overflows, such as by wrapping your Deserializer in the dynamically
+    /// growing stack adapter provided by the serde_stacker crate. Additionally
+    /// you will need to be careful around other recursive operations on the
+    /// parsed result which may overflow the stack after deserialization has
+    /// completed, including, but not limited to, Display and Debug and Drop
+    /// impls.
+    ///
+    /// *This method is only available if serde_json is built with the
+    /// `"unbounded_depth"` feature.*
+    ///
+    /// # Examples
+    ///
+    /// ```edition2018
+    /// use serde::Deserialize;
+    /// use serde_json::Value;
+    ///
+    /// fn main() {
+    ///     let mut json = String::new();
+    ///     for _ in 0..10000 {
+    ///         json = format!("[{}]", json);
+    ///     }
+    ///
+    ///     let mut deserializer = serde_json::Deserializer::from_str(&json);
+    ///     deserializer.disable_recursion_limit();
+    ///     let deserializer = serde_stacker::Deserializer::new(&mut deserializer);
+    ///     let value = Value::deserialize(deserializer).unwrap();
+    ///
+    ///     carefully_drop_nested_arrays(value);
+    /// }
+    ///
+    /// fn carefully_drop_nested_arrays(value: Value) {
+    ///     let mut stack = vec![value];
+    ///     while let Some(value) = stack.pop() {
+    ///         if let Value::Array(array) = value {
+    ///             stack.extend(array);
+    ///         }
+    ///     }
+    /// }
+    /// ```
+    #[cfg(feature = "unbounded_depth")]
+    pub fn disable_recursion_limit(&mut self) {
```

Comment:
Since we anyway go for adding a dedicated method, I would probably accept `depth: Option<u8>` or `depth: Option<usize>` instead for flexibility, but at least being able to disable a hardcoded constant depth is already good enough.

Reply:
I think disabling serde_json's limit is the only essential change. A different recursion limit could be imposed through an adapter outside of serde_json. I filed https://github.com/dtolnay/serde-stacker/issues/1 to include that in serde_stacker.

Reply:
It could, I just don't necessarily see a need for +1 dependency when there is already a field which could be exposed for users who merely want to change the limit.

Reply:
Another alternative would be removing magic constant altogether and leave any stack guards to the `serde-stacker`.

After all, the stack exhaustion issue is not specific to JSON in any way as you mentioned yourself, and I don't see why it has to be handled within this specific adapter.

Reply:
@dtolnay Just in case this got lost - how does this sound / what are your plans about this PR?